File: tests/conformance/fixtures/go/11-full-options.go
package main

import sykli "github.com/yairfalse/sykli/sdk/go"

func main() {
	p := sykli.New()

	p.Task("test").Run("npm test").
		Container("node:20").
		Workdir("/app").
		Env("NODE_ENV", "test").
		Env("CI", "true").
		Retry(3).
		Timeout(300).
		Secrets("NPM_TOKEN", "GH_TOKEN").
		When("branch:main")

	p.Emit()
}

File: tests/conformance/fixtures/elixir/11-full-options.exs
use Sykli

pipeline do
  task "test" do
    run "npm test"
    container "node:20"
    workdir "/app"
    env "CI", "true"
    env "NODE_ENV", "test"
    retry 3
    timeout 300
    secrets ["NPM_TOKEN", "GH_TOKEN"]
    when_ "branch:main"
  end
end
|> Sykli.Emitter.validate!()
|> Sykli.Emitter.to_json()
|> IO.puts()
